packages/typescript-estree/src/node-utils.ts

```diff
@@ -122,8 +122,12 @@ export function isLogicalOperator<T extends ts.SyntaxKind>(
  * @param kind the token's SyntaxKind
  * @returns the token applicable token as a string
  */
-export function getTextForTokenKind(kind: ts.SyntaxKind): string | undefined {
-  return TOKEN_TO_TEXT[kind];
+export function getTextForTokenKind<
+  T extends ts.SyntaxKind,
+  TT = typeof TOKEN_TO_TEXT
+>(kind: T): T extends keyof TT ? TT[T] : undefined {
```

Comment:
do you need the two generics here? Could this be simplified to only use `T`?

Reply:
i can change it to
```suggestion
export function getTextForTokenKind<T extends ts.SyntaxKind>(
  kind: T,
): T extends keyof typeof TOKEN_TO_TEXT ? typeof TOKEN_TO_TEXT[T] : undefined {}
```